### PR TITLE
Add debug layer rendering functionality

### PR DESCRIPTION
**Pull request by reichlfl**
Adds a toggle to visualize all render groups that have a stencil
restriction to debug rendering issues:

image

### CODE REVIEW

File: src/OrbitGl/CaptureWindow.cpp [hunk 1]
```diff
@@ -814,3 +821,41 @@ void CaptureWindow::RenderSelectionOverlay() {
...
+    PrepareGlState();
+    if (time_graph_ != nullptr) {
+      const Color color = time_graph_->GetColor(static_cast<uint32_t>(
+          std::hash<std::string>()(group.name) % std::numeric_limits<uint32_t>::max()));
+      glColor4f(static_cast<float>(color[0]) / 255.f, static_cast<float>(color[1]) / 255.f,
+                static_cast<float>(color[2]) / 255.f, 1.f);
+    } else {
+      glColor4f(1, 0, 0, 1);
+    }
+    glBegin(GL_POLYGON);
```
Comment (pierricgimmig) [inline src/OrbitGl/CaptureWindow.cpp:843]: could we use the batcher here instead of glbegin/glend?

**Review by pierricgimmig — APPROVED** [review]

**Review by florian-kuebler — APPROVED** [review]
lgtm

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/OrbitGl/include/OrbitGl/CaptureWindow.h [hunk 2]
```diff
@@ -90,6 +90,8 @@ class CaptureWindow : public GlCanvas, public orbit_gl::CaptureWindowDebugInterf
   std::unique_ptr<TimeGraph> time_graph_ = nullptr;
   bool draw_help_;
 
+  uint last_rendered_layers_ = 0;
```
[the remark this answers is not in the document]
Comment (reichlfl) [reply src/OrbitGl/include/OrbitGl/CaptureWindow.h:93]: This is indeed no longer needed... thanks, I'll remove it.

File: src/OrbitGl/CaptureWindow.cpp (hunk 1 above)
Comment (reichlfl) [reply src/OrbitGl/CaptureWindow.cpp:843]: Hm I think that would make things more complicated than needed - the Batcher is still used in this mode to collect all primitives, but I'm omitting the render calls. I'd rather keep this debug functionality out of the batcher and keep it like this, the CaptureWindow / GlCanvas already uses raw OpenGl (e.g. to prepare the viewport and state), so I'd argue it's not outside of their scope. WDYT?

**Pull request merged**